### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by carlosmuerto**
# React capstone project - Metrics webapp

**Interactions**
- Home page
  - When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
  - Along with the category name, you will display some numeric values per category that come from the API.
  - When the user clicks (or taps) on a category item, the application navigates to the details page.

- Details page
  - In the details page, the webapp retrieves data from the API to show detailed data in the category.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**
- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
  - You may need to mock the access to the API, so that your tests don't send actual requests.
  - You may need to mock the connection to the Redux Store.

**Technical requirements**

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project should be deployed and accessible online (add a link to the online version of your app to your README file).

[FULL README](https://github.com/microverseinc/curriculum-react-redux/blob/main/capstone/react_capstone.md)

### CODE REVIEW

File: src/setupTests.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
+// allows you to do things like:
+// expect(element).toHaveTextContent(/react/i)
+// learn more: https://github.com/testing-library/jest-dom
+import '@testing-library/jest-dom/extend-expect';
```
Comment (Lordkaito) [inline src/setupTests.js:5]: - [x] As part of the requirements, please make sure to add some tests for your app. Right now I'm not able to find those tests, so please make sure to include them for your next review!

**Review by Lordkaito — CHANGES_REQUESTED** [review]
Hi @carlosmuerto, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.test.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    );
+
+    await waitFor(
+      () => expect(screen.getByText(/classes/i)).toBeInTheDocument(),
+      { timeout: 5000 },
+    );
+    await waitFor(() => expect(screen.getByText(/sets/i)).toBeInTheDocument());
+    await waitFor(() => expect(screen.getByText(/standard/i)).toBeInTheDocument());
+  });
+});
```
Comment (medaminedev66) [inline src/App.test.js:33]: - [x] Following the project requirements, your app should also have unit tests for some pure functions in Redux code. Kindly implement that.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @carlosmuerto 😃,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Your repo looks professional.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/features/Info/HearthstoneInfoSlice.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  it('should handle decrement', () => {
+    const actual = counterReducer(initialState, decrement());
+    expect(actual.value).toEqual(2);
+  });
+
+  it('should handle incrementByAmount', () => {
+    const actual = counterReducer(initialState, incrementByAmount(2));
+    expect(actual.value).toEqual(5);
+  });
+  */
```
Comment (codepantha) [inline src/features/Info/HearthstoneInfoSlice.test.js:51]: - [x] [OPTIONAL] Please remove commented code as it makes your work look cluttered 👍

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @carlosmuerto 👋, 

Well done for implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**